File: csharp/2006/the-illusionist-add.cs
using System;
using System.Threading.Tasks;
using Amazon.DynamoDBv2;
using Amazon.DynamoDBv2.Model;
using System.Collections.Generic;

namespace MovieDatabase
{
    class Program
    {
        static async Task Main(string[] args)
        {
            // Create a MovieRepository instance
            var movies = new MovieRepository();

            // Add "The Illusionist" to the database
            // This demonstrates how to insert a new item into DynamoDB
            await movies.InsertAsync(
                title: "The Illusionist",
                year: 2006,
                plot: "In turn-of-the-century Vienna, a magician uses his abilities to secure the love of a woman far above his social standing.",
                rating: 7.6
            );

            // Confirm that the movie was added by retrieving it
            var movie = await movies.SelectAsync(
                title: "The Illusionist",
                year: 2006
            );

            if (movie != null)
            {
                // The movie was found
                Console.WriteLine($"Movie found: {movie}");
            }
            else
            {
                // The movie was not found
                Console.WriteLine("Movie not found");
            }
        }
    }
}

File: python/2006/the-illusionist-add.py

from movies import MovieRepository

# Create a new movie repository using the default configuration
movies = MovieRepository()

# Add The Illusionist to the database
movies.insert(
    title="The Illusionist", 
    year=2006, 
    plot="In turn-of-the-century Vienna, a magician uses his abilities to secure the love of a woman far above his social standing.", 
    rating=7.6
)

# Confirm that the movie was added
movie = movies.select(
    title="The Illusionist", 
    year=2006
)
if movie:
    # The movie was found
    print(f"Movie found: {movie}")
else:
    # The movie was not found
    print("Movie not found")
    